tower-test/src/mock/future.rs

```diff
@@ -23,18 +28,17 @@ impl<T> ResponseFuture<T> {
 }
 
 impl<T> Future for ResponseFuture<T> {
-    type Item = T;
-    type Error = Error;
+    type Output = Result<T, Error>;
 
-    fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
+    fn poll(mut self: Pin<&mut Self>, cx: &mut Context) -> Poll<Self::Output> {
         match self.rx {
-            Some(ref mut rx) => match rx.poll() {
-                Ok(Async::Ready(Ok(v))) => Ok(v.into()),
-                Ok(Async::Ready(Err(e))) => Err(e),
-                Ok(Async::NotReady) => Ok(Async::NotReady),
-                Err(_) => Err(error::Closed::new().into()),
+            Some(ref mut rx) => match Box::pin(rx).as_mut().poll(cx) {
```

Comment:
It can be written as follows by using [`pin_utils::unsafe_pinned!(rx: Option<Rx<T>>)`](https://docs.rs/pin-utils/0.1.0-alpha.4/pin_utils/macro.unsafe_pinned.html) and [`Option::as_pin_mut()`](https://doc.rust-lang.org/nightly/core/option/enum.Option.html#method.as_pin_mut):
```rust
    match self.rx().as_pin_mut() {
        Some(rx) => match rx.poll(cx) {
        // ...
    
```

Reply:
Done!